plugins/polio/js/src/domains/Campaigns/Scope/ScopeForm.tsx

```diff
@@ -77,44 +77,47 @@ export const ScopeForm: FunctionComponent = () => {
     }, [currentTab, rounds, scopePerRound, sortedRounds, values.scopes]);
 
     const filteredDistricts: FilteredDistricts[] | undefined = useMemo(() => {
-        if (districtShapes && regionShapes) {
-            let filtered: FilteredDistricts[] = districtShapes.map(district => {
-                const scope = findScopeWithOrgUnit(scopes, district.id);
-                return {
-                    ...cloneDeep(district),
-                    region: findRegion(district, regionShapes),
-                    scope,
-                    vaccineName: scope?.vaccine,
-                };
-            }) as FilteredDistricts[];
-            if (scopes && isPolio) {
-                filtered.forEach((d, index) => {
-                    scopes.forEach(scope => {
-                        scope.group.org_units.forEach(ouId => {
-                            if (d.id === ouId) {
-                                filtered[index].vaccineName = scope.vaccine;
-                            }
-                        });
-                    });
-                });
-            }
+        if (!districtShapes || !regionShapes) return undefined;
 
-            if (searchScope) {
-                filtered = filtered.filter(d =>
-                    scopes.some(scope => scope.group.org_units.includes(d.id)),
-                );
-            }
+        const orgUnitIdToVaccine = new Map();
+        if (isPolio && scopes) {
+            scopes.forEach(scope => {
+                scope.group.org_units.forEach(ouId => {
+                    orgUnitIdToVaccine.set(ouId, scope.vaccine);
+                });
+            });
+        }
```

Comment:
If we ever remake the API, it would be great to have that map coming straight from the backend